olive/resource_path.py

```diff
@@ -316,64 +317,126 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
+            )
 
-        # azureml client
-        ml_client = self.config.azureml_client.create_client()
+        dir_path = Path(dir_path).resolve()
+        dir_path.mkdir(parents=True, exist_ok=True)
 
-        # create aml model
-        logger.debug(f"Creating aml model for datastore {self.config.datastore_name} path {self.config.relative_path}.")
-        aml_model = retry_func(
-            ml_client.models.create_or_update,
-            [
-                Model(
-                    path=self.get_path(),
-                    name="olive-backend-model",
-                    description="Model created by Olive backend. Ignore this model.",
-                    type=AssetTypes.CUSTOM_MODEL,
-                )
-            ],
-            max_tries=self.config.azureml_client.max_operation_retries,
-            delay=self.config.azureml_client.operation_retry_interval,
-            exceptions=ServiceResponseError,
-        )
+        self.get_aml_client_config()
+
+        # azureml file system
+        fs = AzureMachineLearningFileSystem(self.get_path())
+        relative_path = Path(self.get_relative_path())
+        is_file = self.is_file(fs)
+        # path to save the resource to
+        if name:
+            new_path_name = Path(name).with_suffix("" if not is_file else relative_path.suffix).name
+        else:
+            new_path_name = relative_path.name
 
-        # use the AzureMLModel to download the model
+        new_path = dir_path / new_path_name
+        _overwrite_helper(dir_path / new_path_name, overwrite)
+
+        # download artifacts to a temporary directory
         logger.debug(
-            f"Downloading aml model for datastore {self.config.datastore_name} path {self.config.relative_path}."
-        )
-        azureml_model_resource = AzureMLModel(
-            {"azureml_client": self.config.azureml_client, "name": aml_model.name, "version": aml_model.version}
+            f"Downloading aml resource for datastore {self.config.datastore_name} path {self.config.relative_path}."
         )
-        return azureml_model_resource.save_to_dir(dir_path, name, overwrite)
+
+        with tempfile.TemporaryDirectory(dir=dir_path, prefix="olive_tmp") as temp_dir:
+            retry_func(
+                fs.download,
+                kwargs={
+                    "rpath": self.get_relative_path(),
+                    "lpath": temp_dir,
+                    "recursive": not is_file,
+                },
+                max_tries=self.config.azureml_client.max_operation_retries,
+                delay=self.config.azureml_client.operation_retry_interval,
+            )
+            downloaded_resource = Path(temp_dir) / relative_path.name
+            if downloaded_resource.is_file():
+                # only if the resource is a existed file we will move it to the new path
+                shutil.move(downloaded_resource, new_path)
+            else:
+                shutil.copytree(temp_dir, new_path)
```

Comment:
not a big issue but would shutil.move work for this part too? If so, I think that might be better/faseter since shutil.move  will be just a rename. the temp_dir is in the same partition as `dir_path` so there is no copy done in move. 

something like
```
new_path.parent.mkdir(parents=True, exist_ok=True)
shutil.move(downloaded_resource  is is_file else temp_dir, new_path)
``` 
?